### DISCUSSION TITLE
Why can't I use ISR for dynamic routes

### DISCUSSION THREAD

**Discussion by hornta** [vercel/next.js#36528] [Help]
I would love to use ISR for dynamic routes so that I can cache all my item pages for all users instead of relying on the cache-control header. My userbase is mostly visiting 1-2 pages a session so the cache header doesn't really solve my problems. 

Most of the dynamic pages content updates like once a week so I'm looking at caching it serverside, but the problem is that I can't access dynamic route segment from the `getStaticProps` function.

Do you have any ideas?

**Comment by ijjk** [comment]
Hi, the dynamic route segment is available under the `params` in the context passed to `getStaticProps`, the related docs for this can be seen here https://nextjs.org/docs/api-reference/data-fetching/get-static-props#context-parameter
